lib/Migration/Version010000001Date20250727094721.php

```diff
@@ -69,10 +71,13 @@ public function postSchemaChange(IOutput $output, Closure $schemaClosure, array
 		$query = $this->db->getQueryBuilder();
 		$query->update('recognize_face_detections')
 			->set('face_vector', 'vector')
-			->where($query->expr()->in('id', $query->createFunction('(' . $select->getSQL() .')')));
+			->where($query->expr()->in('id', $query->createParameter('ids')));
```

Comment:
Shouldn't you be able to update all with a single query?
```suggestion
			->where($select->expr()->isNull('face_vector'));
		$query->executeStatement()
```
if you are not doing any data manipulation that should work just fine.
No previous select or anything needed.

Reply:
> Shouldn't you be able to update all with a single query?

Doesn't seem work for large tables, sadly. See #1357

Reply:
Mh, the issue in #1357 is due to a long running transaction, wouldn't wrapping the current state in this branch in a transaction introduce the issue of a long running transaction again?

Reply:
You can close the transaction after each chunk.
THe idea is to reduce the number of writes